lib/features/splash/cubit/splash_cubit.dart

```diff
@@ -9,39 +14,86 @@ class SplashCubit extends CommonCubit<void, SplashCustom> {
   SplashCubit(
     this._authRepository,
     this._profilesRepository,
+    this._networkInfo,
   ) : super(const BaseState.loading());
 
   final FamilyAuthRepository _authRepository;
   final ProfilesRepository _profilesRepository;
+  final NetworkInfo _networkInfo;
+  StreamSubscription<bool>? _internetConnectionSubscription;
 
   Future<void> init() async {
-    await _authRepository.initAuth();
-    final user = _authRepository.getCurrentUser();
-    final profiles = await _profilesRepository.refreshProfiles();
-
-    if (user == null) {
-      emitCustom(const SplashCustom.redirectToWelcome());
-      return;
+    _internetConnectionSubscription = _networkInfo
+        .hasInternetConnectionStream()
+        .listen((hasInternetConnection) async {
+      if (hasInternetConnection) {
+        await _checkForRedirect();
+      } else {
+        _showNoInternetMessage();
+      }
+    });
+    if (_networkInfo.isConnected) {
+      await _checkForRedirect();
+    } else {
+      _showNoInternetMessage();
     }
+  }
 
-    if (false == user.isUsUser) return;
+  void _showNoInternetMessage() {
+    emitCustom(const SplashCustom.noInternet());
+    emitLoading();
+  }
 
-    final fbsdk = FacebookAppEvents();
-    await fbsdk.setAutoLogAppEventsEnabled(true);
-    await fbsdk.logEvent(
-      name: 'app_open_and_logged_in',
-    );
+  Future<void> _checkForRedirect() async {
+    try {
+      await _authRepository.initAuth();
+      Future.delayed(
+        const Duration(seconds: 2),
+        () => NotificationService.instance.scheduleBedtimeTestNotifications(),
```

Comment:
Should this always be scheduled? Or is there a check on debug/test somewhere?

Reply:
oopsie, removed it! was test code